license.go

```diff
@@ -225,3 +218,38 @@ func (l *License) GuessType() error {
 func scan(text, match string) bool {
 	return strings.Contains(text, match)
 }
+
+// returns a []string of files in a directory, or error
+func readDirectory(dir string) ([]string, error) {
+	d, err := os.Stat(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	if !d.IsDir() {
+		return nil, fmt.Errorf("license: cannot search %s: not a directory", dir)
+	}
+
+	fileinfos, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+	files := make([]string, len(fileinfos))
+	for pos, fi := range fileinfos {
+		files[pos] = fi.Name()
+	}
+	return files, nil
+}
+
+// returns the first file that case-insensitive matches any of the
+// license files
+func matchLicenseFile(licenses []string, files []string) string {
+	for _, file := range files {
+		for _, license := range licenses {
+			if strings.EqualFold(license, file) {
+				return file
```

Comment:
Now that I'm thinking about this more, we should probably report if there is more than a single license file found that we recognize, to avoid reporting the wrong license type.